src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -145,6 +145,10 @@ public override Binder VisitMethodDeclaration(MethodDeclarationSyntax methodDecl
                     {
                         method = method ?? GetMethodSymbol(methodDecl, resultBinder);
                         resultBinder = new InMethodBinder(method, resultBinder);
+                        if (method.IsAsync)
```

Comment:
Expected to see `&& method.IsIterator`

Reply:
`IsIterator` only gives a correct result after we've bound the body of the method (ie. detected `yield`). 

Two options:
- We could consider adding a syntax-based detection of iterators
- The logic inside this new binder checks for `async` + `IAsyncEnumerable`. I could move that check out here instead.

Reply:
This part of the code is less familiar to me so I'd tend to trust you / other judgement here.